pkg/utils/utils.go

```diff
@@ -0,0 +1,26 @@
+package utils
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// HasFinalizer returns true if the given object has the given finalizer
```

Comment:
Don't need to hand-roll these functions, should import them from `"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"`

e.g. [ContainsFinalizer](https://github.com/kubernetes-sigs/controller-runtime/blob/92646a5615783bfa9d8196ae4571150742e7572d/pkg/controller/controllerutil/controllerutil.go#L380), [AddFinalizer](https://github.com/kubernetes-sigs/controller-runtime/blob/92646a5615783bfa9d8196ae4571150742e7572d/pkg/controller/controllerutil/controllerutil.go#L353), and [RemoveFinalizer](https://github.com/kubernetes-sigs/controller-runtime/blob/92646a5615783bfa9d8196ae4571150742e7572d/pkg/controller/controllerutil/controllerutil.go#L366)